unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/AudioSource/ECSAudioSourceComponentHandler.cs

```diff
@@ -0,0 +1,194 @@
+        {
+            if(promiseAudioClip != null)
+                AssetPromiseKeeper_AudioClip.i.Forget(promiseAudioClip);
+            
+            CommonScriptableObjects.sceneID.OnChange -= OnCurrentSceneChanged;
+
+            if (Settings.i != null)
+                Settings.i.audioSettings.OnChanged -= OnAudioSettingsChanged;
+            
+            DataStore.i.virtualAudioMixer.sceneSFXVolume.OnChange -= OnVirtualAudioMixerChangedValue;
+            if (audioSource != null)
+            {
+                GameObject.Destroy(audioSource);
+                audioSource = null;
+            }
+        }
+        
+        public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, ECSAudioSource model)
+        {
+            bool isSameClip = model.audioClipUrl == this.model?.audioClipUrl;
+            this.scene = scene;
+            this.model = model;
+            
+            ApplyCurrentModel();
+            
+            // If the clip has changed, we need to forget the old clip
+            if(!isSameClip && promiseAudioClip != null)
+                AssetPromiseKeeper_AudioClip.i.Forget(promiseAudioClip);
+            
+            promiseAudioClip = new AssetPromise_AudioClip(model.audioClipUrl, scene.contentProvider);
+            promiseAudioClip.OnSuccessEvent += OnComplete;
+            promiseAudioClip.OnFailEvent += OnFail;
+
+            AssetPromiseKeeper_AudioClip.i.Keep(promiseAudioClip);
+            
+            CommonScriptableObjects.sceneID.OnChange -= OnCurrentSceneChanged;
+            CommonScriptableObjects.sceneID.OnChange += OnCurrentSceneChanged;
+        }
+        
+        private void ApplyCurrentModel()
+        {
+            if (audioSource == null)
+            {
+                Debug.LogWarning("AudioSource is null!.");
+                return;
+            }
+            
+            UpdateAudioSourceVolume();
+            audioSource.loop = model.loop;
+            audioSource.pitch = model.pitch;
+            audioSource.spatialBlend = 1;
+            audioSource.dopplerLevel = 0.1f;
+
+            if (!model.playing)
+            {
+                if (audioSource.isPlaying)
+                {
+                    audioSource.Stop();
+                }
+            }
+        }
+        
+        private void ApplyLoadedAudioClip(AudioClip clip)
+        {
+            if (audioSource.clip != clip)
+                audioSource.clip = clip;
+            
+            bool shouldPlay = playedAtTimestamp != model.playedAtTimestamp ||
+                              (model.playing && !audioSource.isPlaying);
+
+            //To remove a pesky and quite unlikely warning when the audiosource is out of scenebounds
+            if (audioSource.enabled && model.playing && shouldPlay)
+                audioSource.Play();
+            
+            playedAtTimestamp = model.playedAtTimestamp;
+        }
+        
+        public void UpdateOutOfBoundariesState(bool isInsideBoundaries)
```

Comment:
could we implement this interface's method explicitly?
```suggestion
        void IOutOfSceneBoundariesHandler.UpdateOutOfBoundariesState(bool isInsideBoundaries)
```

Reply:
I like it more like a public method but I will change it